src/hotspot/share/code/compiledMethod.cpp

```diff
@@ -478,6 +478,8 @@ bool CompiledMethod::clean_ic_if_metadata_is_dead(CompiledIC *ic) {
       } else {
         ShouldNotReachHere();
       }
+    } else {
+      return true;
```

Comment:
I've given up pretending to understand this code, but could you add a one line comment why you're returning true here?  ie. if ic_metadata is NULL, it's a megamorphic call or already clean and shouldn't be cleaned.

Reply:
Thanks for the comment.